Fix test failures found in Debian builds

jamessan opened the pull request:
Closes #7522

jamessan committed:
tui_spec: Convert nil to "" before formatting it

This fixes an apparent difference in behavior between Lua and LuaJIT.
Lua fails to format nil:

    test/functional/terminal/tui_spec.lua:381: bad argument #2 to 'format' (string expected, got nil)

jamessan committed:
Use PRId64 to format Integer when calling api_set_error

Integer is a 64-bit type so using %d can produce incorrect results.

    test/functional/api/highlight_spec.lua @ 35: highlight api nvim_get_hl_by_id
    ...W7Xi/neovim-0.2.1/test/functional/api/highlight_spec.lua:46: Expected objects to be the same.
    Passed in:
    (string) 'Invalid highlight id: 7671724'
    Expected:
    (string) 'Invalid highlight id: 30000'

bfredl commented on test/functional/terminal/tui_spec.lua:
```diff
@@ -380,7 +380,7 @@ describe("tui 't_Co' (terminal colors)", function()
     screen = thelpers.screen_setup(0, string.format(
       [=[['sh', '-c', 'LANG=C TERM=%s %s %s -u NONE -i NONE --cmd "silent set noswapfile noshowcmd noruler"']]=],
-      term,
+      term and term or "",
```
Is this not just `term or ""` ?

jamessan committed:
tests: terminal: Assert for SIGWINCH handling before continuing

Fixes test failures like

    test/functional/terminal/cursor_spec.lua @ 62: terminal cursor with number column is positioned correctly when focused
    ./test/functional/ui/screen.lua:302: Row 2 did not match.
    Expected:
      |{7:  1 }tty ready                                     |
      |*{7:  2 }{1: }                                             |
      |{7:  3 }                                              |
      |{7:  4 }                                              |
      |{7:  5 }                                              |
      |{7:  6 }                                              |
      |{3:-- TERMINAL --}                                    |
    Actual:
      |{7:  1 }tty ready                                     |
      |*{7:  2 }rows: 6, cols: 46                             |
      |{7:  3 }{1: }                                             |
      |{7:  4 }                                              |
      |{7:  5 }                                              |
      |{7:  6 }                                              |
      |{3:-- TERMINAL --}                                    |

ZyX-I commented:
Most of project-local integers have defines like `PRIdNUMBER`, `PRIdSCID`, `PRIdLINENR`, …; new define should be created when needed in place of using PRId64, or number should be cast.
